src/services/DocumentVerifier.ts

```diff
@@ -1,52 +1,36 @@
-import { verifyWithIndividualChecks } from "@govtechsg/oa-verify";
-import { SignedDocument } from "@govtechsg/open-attestation";
-import { checkValidIdentity } from "./IdentityVerifier";
+import {
+  verificationBuilder,
+  openAttestationVerifiers,
+  VerificationManagerOptions,
+  VerificationFragmentStatus
+} from "@govtechsg/oa-verify";
+import { OAWrappedDocument } from "../types";
 
-// Ensures TS infers the type of the array as a tuple instead
-// https://github.com/Microsoft/TypeScript/pull/24897
-function tuple<T extends any[]>(...data: T): T {
-  return data;
-}
+const verifiers = [
+  ...openAttestationVerifiers.slice(0, 2),
+  ...openAttestationVerifiers.slice(3, 5)
+];
+const defaultVerify = verificationBuilder(verifiers);
 
-// Let TS infer the return type
-// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export const checkValidity = (
-  document: SignedDocument,
-  network = "ropsten" // TODO: handle this based on some user selection of the network
-) => {
-  const isMainnet = network === "mainnet";
-  const networkName = isMainnet ? "homestead" : "ropsten";
+export const checkValidity = async (
+  document: OAWrappedDocument,
+  network = "ropsten",
+  promisesCallback: VerificationManagerOptions["promisesCallback"],
+  verify = defaultVerify
+): Promise<VerificationFragmentStatus> => {
+  const overallResult = await verify(document, {
+    network,
+    promisesCallback
+  });
 
-  const [verifyHash, verifyIssued, verifyRevoked] = verifyWithIndividualChecks(
-    document,
-    networkName
-  );
-
-  const verifyIdentity = checkValidIdentity(document, networkName);
-
-  // If any of the checks are invalid, resolve the overall validity early
-  const overallValidityCheck = Promise.all([
-    new Promise(async (resolve, reject) =>
-      (await verifyHash).checksumMatch ? resolve() : reject()
-    ),
-    new Promise(async (resolve, reject) =>
-      (await verifyIssued).issuedOnAll ? resolve() : reject()
-    ),
-    new Promise(async (resolve, reject) =>
-      (await verifyRevoked).revokedOnAny ? reject() : resolve()
-    ),
-    new Promise(async (resolve, reject) =>
-      (await verifyIdentity).identifiedOnAll ? resolve() : reject()
-    )
-  ])
-    .then(() => true)
-    .catch(() => false);
-
-  return tuple(
-    verifyHash,
-    verifyIssued,
-    verifyRevoked,
-    verifyIdentity,
-    overallValidityCheck
-  );
+  const statuses = overallResult.map(f => f.status);
```

Comment:
oa-verify export this: https://github.com/Open-Attestation/oa-verify#isvalid

Reply:
Hmm yea, thing is I wanted none of the tests to skip (ISSUER_IDENTITY was being skipped for one of our tests files), so in the case that a test skips, then it should return `ERROR`.

Reply:
Will use `isValid` for now and handle error states in future perhaps within `oa-verify`